src/Sarif.FunctionalTests/ConverterTestData/StaticDriverVerifier/checkadddevice_bug1.tt.sarif

```diff
@@ -32,154 +32,105 @@
             {
               "locations": [
                 {
+                  "fullyQualifiedLogicalName": "OS",
                   "kind": "call",
+                  "callee": "main",
                   "importance": "unimportant",
                   "properties": {
-                    "State": "^",
-                    "Caller": "OS",
-                    "Callee": "main"
+                    "currentDataValues": "^"
                   }
                 },
                 {
                   "step": 1,
-                  "message": "====Auto=====",
                   "kind": "continuation",
-                  "importance": "unimportant",
-                  "properties": {
-                    "State": "^====Auto====="
-                  }
+                  "importance": "unimportant"
                 },
                 {
                   "step": 2,
-                  "message": "====Auto=====,====Pre=====,____true",
+                  "fullyQualifiedLogicalName": "main",
                   "kind": "call",
+                  "callee": "_sdv_init4",
                   "importance": "unimportant",
                   "properties": {
-                    "State": "^====Auto=====^====Pre=====^____true",
-                    "Caller": "main",
-                    "Callee": "_sdv_init4"
+                    "currentDataValues": "^====Pre=====^____true"
                   }
                 },
                 {
                   "step": 3,
-                  "message": "====Auto=====",
                   "kind": "continuation",
-                  "importance": "unimportant",
-                  "properties": {
-                    "State": "^====Auto====="
-                  }
+                  "importance": "unimportant"
                 },
                 {
                   "step": 4,
-                  "message": "====Auto=====",
                   "kind": "assignment",
-                  "importance": "unimportant",
-                  "properties": {
-                    "State": "^====Auto====="
-                  }
+                  "importance": "unimportant"
                 },
                 {
                   "step": 5,
-                  "message": "====Auto=====",
                   "kind": "assignment",
-                  "importance": "unimportant",
-                  "properties": {
-                    "State": "^====Auto====="
-                  }
+                  "importance": "unimportant"
                 },
                 {
                   "step": 6,
-                  "message": "====Auto=====",
+                  "fullyQualifiedLogicalName": "main",
```

Comment:
ok, happily, I was not correct, functionEnter and functionExit survived. So, call and callReturn refer to the location of a call site. functionEnter and functionExit are the matched pair of locations within a callee. FQLN should be identical for each pair.

btw - the spec does not make it clear whether we require these to be emitted as matched pairs or not.